moleculekit/smallmol/smallmol.py

```diff
@@ -756,6 +756,102 @@ def write(self, fname, frames=None, merge=True):
     def view(self, *args, **kwargs):
         self.toMolecule().view(*args, **kwargs)
 
+    def describe(self):
+        """
+        Returns rdkit descriptors for the molecule, like DESC_NumRotatableBonds or DESC_MolLogP. See rdkit.Chem.Descriptors for more.
+
+        Returns
+        -------
+        smart: str
+            The smarts string
+        """
+        from rdkit.Chem import Descriptors
+
+        desc_dict = dict(Descriptors.descList)
+        descs = list(desc_dict.keys())
+        descs.remove('Ipc')
+        ans = {}
+        for descname in descs:
+            try:
+                desc = desc_dict[descname]
+                value = desc(self._mol)
+            except (ValueError, TypeError, ZeroDivisionError) as exception:
+                logger.warning(f"Descriptor {descname} was not calculated for this small molecule due to {str(exception)}")
+                value = 'NaN'
+
+            bin_name = 'DESC_{}'.format(descname)
+            ans[bin_name] = value
+
+        molstring = ans.values()
+        headers = ans.keys()
+        return molstring, headers
+
+    def fingerprint(self, radius=2, num_bits=1024):
+        """
+        Returns Morgan, MACCS and AvalonCount fingerprints at specified radius and num_bits
+
+        Parameters
+        ----------
+        radius: int
+           Radius to define a local environment for the relevant fingeprints
+        num_bits: int
+            The number of bits to use in the fingerprint. Larger avoids collisions.
+
+        Returns
+        -------
+        A list with the three fingerprints
+        """
+        from rdkit.Avalon.pyAvalonTools import GetAvalonCountFP
+        from rdkit.Chem import MACCSkeys
+        from rdkit.Chem import AllChem
+
+        ecfp4 = AllChem.GetHashedMorganFingerprint(self._mol, radius, num_bits)
+        maccs = MACCSkeys.GenMACCSKeys(self._mol)
+        avalons = GetAvalonCountFP(self._mol, num_bits)
+        return (ecfp4, maccs, avalons)
```

Comment:
no need for the parentheses. doesn't matter really (except for taking up space) but just fyi